semantic_release/cli/config.py

```diff
@@ -145,7 +145,7 @@ def resolve_env_vars(cls, val: Any) -> str | None:
     @model_validator(mode="after")
     def set_default_token(self) -> Self:
         # Set the default token name for the given VCS when no user input is given
-        if not self.token and self.type in _known_hvcs:
+        if not self.token and self.type in _known_hvcs:  # noqa: SIM102
```

Comment:
```suggestion
        if self.token:
            return self
        if self.type not in _known_hvcs:
            return self
        if env_token := self._get_default_token():
            self.token = env_token
        return self
```

Reply:
This is probably what it should of been in the first place, following never nesting practice.